prboom2/src/SDL/i_video.c

```diff
@@ -701,23 +711,29 @@ static void I_FillScreenResolutionsList(void)
   list_size = 0;
   current_resolution_index = -1;
 
+  // on success, SDL_GetNumDisplayModes() always returns at least 1
   if (count > 0)
   {
-    count = MIN(count, MAX_RESOLUTIONS_COUNT - 2);
+    // -2 for the desired resolution and for NULL
+    count = MIN(count, MAX_RESOLUTIONS_COUNT - 2 - num_canonicals);
 
-    for(i = count - 1; i >= 0; i--)
+    for(i = count - 1 + num_canonicals; i >= 0; i--)
     {
       int in_list = false;
 
-      SDL_GetDisplayMode(display_index, i, &mode);
+      // make sure the canonical resolutions are always available
+      if (i > count - 1)
```

Comment:
Is it possible for the canonical resolutions to already be included in the display modes returned by SDL? Would we end up with duplicate entries in the list?

Reply:
Nope, that's accounted for a few lines down (check for the `in_list` variable). SDL would return all the different *video modes* that your display supports - i.e. all possible combinations of resolution, color depth, refresh rate, etc. - whereas we only take the resolution into consideration. Thus, we already expect the actual resolutions list to be made up of a list of video modes that may contain duplicate resolutions.